packages/cw721/README.md

```diff
@@ -120,14 +136,72 @@ value (suggested 30). Contracts can define other `DefaultLimit` and `MaxLimit`
 values without violating the CW721 spec, and clients should not rely on
 any particular values.
 
-If `start_after` is unset, the query returns the first results, ordered 
+If `start_after` is unset, the query returns the first results, ordered
 lexicographically by `token_id`. If `start_after` is set, then it returns the
-first `limit` tokens *after* the given one. This allows straightforward 
+first `limit` tokens _after_ the given one. This allows straightforward
 pagination by taking the last result returned (a `token_id`) and using it
-as the `start_after` value in a future query. 
+as the `start_after` value in a future query.
 
 `Tokens{owner, start_after, limit}` - List all token_ids that belong to a given owner.
 Return type is `TokensResponse{tokens: Vec<token_id>}`.
 
-`AllTokens{start_after, limit}` - Requires pagination. Lists all token_ids controlled by 
+`AllTokens{start_after, limit}` - Requires pagination. Lists all token_ids controlled by
 the contract.
+
+### NftInfo Extension - CW721 Metadata Onchain
+
+NFT creators may want to store their NFT metadata on-chain so other contracts are able to interact with it.
+With CW721 in CosmWasm, we allow you to store any data on chain you wish, using a generic `extension: T`.
+
+In order to support on-chain metadata, and to demonstrate how to use the extension ability, we have created this simple contract.
+There is no business logic here, but looking at `lib.rs` will show you how do define custom data that is included when minting and
+available in all queries.
+
+In particular, here we define:
+
+```rust
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct Trait {
+    pub display_type: Option<String>,
+    pub trait_type: String,
+    pub value: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct Metadata {
```

Comment:
Should we have some kind of flexible field here if we are putting all these attributes? I wonder if we would ever need more than what's listed here.

Reply:
It is flexible enough. There is `Metadata.attributes` which is a list of `Trait`s:

```rs
#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
pub struct Trait {
    pub display_type: Option<String>,
    pub trait_type: String,
    pub value: String,
}
```

Reply:
So it follows 100% ERC721 spec, using metadata and traits.